### ISSUE TITLE
RCS1146: "Use conditional access" does not trigger if `default` keyword is used instead of `null`

### ISSUE THREAD

**Issue by IanKemp** [JosefPihrt/Roslynator#735]
**Product and Version Used**: 

Microsoft Visual Studio Enterprise 2019
Version 16.7.6

Roslynator 2019
Version 3.0.0

**Steps to Reproduce**:

Write a statement that retrieves a reference-type object instance, compares it to `default`, and if the instance is not null, performs some operation on that instance. Note that RCS1146 is not triggered.

Now replace `default` with `null` (these are the same for reference types) and notice that RCS1146 is triggered.

**Actual Behavior**:
```csharp
var someReferenceObject = GetSomeReferenceObject();
if (someReferenceObject != default && someReferenceObject.BoolProperty) // no RCS1146
{
    ...
}

var someReferenceObject = GetSomeReferenceObject();
if (someReferenceObject != null && someReferenceObject.BoolProperty) // RCS1146
{
    ...
}
```

**Expected Behavior**:
RCS1146 is triggered in both of the above cases and suggests to change the code to:

```csharp
var someReferenceObject = GetSomeReferenceObject();
if (someReferenceObject?.BoolProperty == true)
{
    ...
}
```
